components-core/src/main/java/com/adyen/checkout/components/util/ValidationUtils.kt

```diff
@@ -0,0 +1,56 @@
+package com.adyen.checkout.components.util
+
+import com.adyen.checkout.core.api.Environment
+import java.util.regex.Pattern
+
+object ValidationUtils {
+
+    private const val EMAIL_REGEX = "^[\\w\\.-]+@([\\w\\-]+\\.)+[A-Z]{2,4}$"
+    private val EMAIL_PATTERN = Pattern.compile(ValidationUtils.EMAIL_REGEX, Pattern.CASE_INSENSITIVE)
+
+    private const val PHONE_REGEX = "^\\D*(\\d\\D*){9,14}$"
+    private val PHONE_PATTERN = Pattern.compile(PHONE_REGEX)
+
+    private const val CLIENT_KEY_REGEX = "([a-z]){4}\\_([A-z]|\\d){32}"
+    private val CLIENT_KEY_PATTERN = Pattern.compile(CLIENT_KEY_REGEX)
+
+    private const val CLIENT_KEY_TEST_PREFIX = "test_"
+    private const val CLIENT_KEY_LIVE_PREFIX = "live_"
+
+    /**
+     * Check if phone number is valid.
+     *
+     * @param phoneNumber A string to check if is a phone number.
+     * @return If we consider it a valid phone number or not.
+     */
+    fun isPhoneNumberValid(phoneNumber: String): Boolean {
+        return PHONE_PATTERN.matcher(phoneNumber).matches()
+    }
+
+    /**
+     * Check if email is valid.
+     *
+     * @param emailAddress A string to check if is an email address.
+     * @return If we consider it a valid email or not.
+     */
+    fun isEmailValid(emailAddress: String): Boolean {
+        return EMAIL_PATTERN.matcher(emailAddress).matches()
+    }
+
+    /**
+     * Check if the Client Key is valid.
+     *
+     * @param clientKey A string to check if is a client key.
+     * @param environment Selected environment in the configuration.
+     * @return If we consider it a valid client key or not.
+     */
+    fun isClientKeyValid(clientKey: String, environment: Environment): Boolean {
+        val isLiveEnvironment = environment == Environment.AUSTRALIA || environment == Environment.UNITED_STATES || environment == Environment.EUROPE
```

Comment:
Maybe we should override `.equals` inside `Environment` to make sure it compares URLs and not the whole object? I'm thinking if for some reason the object gets copied and the `==` would not work because the instances are different? (happened to me once with `Amount.EMPTY`, not sure in which scenario).

There is also the case where someone might pass an environment manually like `Environment(URL("SOME_URL"))` and this `"SOME_URL"` matches one of our environments (even though this might be an overreach, not sure anyone does that).

Reply:
This is a fair point, I think it would be easier for us to assume that any environment that's not `Environment.TEST` is a live one.

Reply:
Got it. Removing the case where it's neither test nor live. Should we update the ticket as well?

Reply:
@caiofaustino the ticket does say `For custom environments don't check the clientKey.`. I think Ozgur had it right first, the only issue was that `Environment.equals` was not implemented so it was checking if the instance the exact same one.

Reply:
I doubt anyone is using custom Environments, and if they are, I think it's fair to assume those are LIVE.

Reply:
To be safe I added the custom environment case again.